src/libfuzzer/libfuzzer_macro.h

```diff
@@ -33,6 +33,26 @@
 // significantly slower than mutator, so fuzzing rate may stay unchanged.
 #define DEFINE_BINARY_PROTO_FUZZER(arg) DEFINE_PROTO_FUZZER_IMPL(true, arg)
 
+// Registers the callback as a potential mutation performed on the parent
+// message of a field. This must be called inside an initialization code block.
+// libFuzzer suggests putting one-time-initialization in a function used to
+// initialize a static variable inside the fuzzer target. For example:
+//
+// static bool Callback(protobuf::Message* message) { ... }
+//
+// static bool OneTimeInitialization() {
+//   REGISTER_PROTO_FIELD_MUTATOR(SomeMessage, "name_of_the_field", Callback);
+//   ...
+// }
+//
+// DEFINE_PROTO_FUZZER(const SomeMessage& msg) {
+//   static bool initialized = OneTimeInitialization();
+//   ...
+// }
+#define REGISTER_PROTO_FIELD_MUTATOR(message_class, field_name, callback) \
```

Comment:
That not exactly I had in mind but I don't have answer right now.
I am ok commit this part as-is and later I'll try to improve it.

Reply:
Static initializers look like they would work for this, but they have to work without introducing race conditions.